File: Py/d.records.py
from dataclasses import dataclass

@dataclass(frozen=True)
class Person:
    name: str
    age: int
    email: str = "" 

# Usage
person1 = Person("Alice", 30)
person2 = Person("Alice", 30)
person3 = Person("Bob", 25, "bob@example.com")

print(person1)          # Person(name='Alice', age=30, email='')
print(person1 == person2)  # True (equality based on fields)
print(person3.email)    # bob@example.com

# person1.name = "Eve"  # Raises FrozenInstanceError due to frozen=True

File: C#/D.Records.cs
public record Person(string Name, int Age, string Email = "")
{    
    public override string ToString() => $"{Name}, {Age}, {Email}";
}

// Usage
class Program2
{
    static void Main()
    {
        var person1 = new Person("Alice", 30);
        var person2 = new Person("Alice", 30);
        var person3 = new Person("Bob", 25, "bob@example.com");

        Console.WriteLine(person1);         // Alice, 30, 
        Console.WriteLine(person1 == person2); // True (value equality)
        Console.WriteLine(person3.Email);   // bob@example.com

        // With-expression for immutable updates
        var person4 = person1 with { Age = 31 };
        Console.WriteLine(person4);         // Alice, 31, 
    }
}